apps/studio/src/services/navigation.service.ts

```diff
@@ -28,9 +28,18 @@ export class NavigationService extends AbstractService {
     hash: string,
   ) {
     try {
-      const range = this.svcs.parserSvc.getRangeForJsonPath('asyncapi', jsonPointer);
+      const doc = this.svcs.editorSvc;
+      const docType = doc.value.startsWith('asyncapi') ? 'yaml' : 'json';
+
+      let range;
+      if (docType === 'yaml') {
+        range = this.svcs.parserSvc.getRangeForYamlPath('asyncapi', jsonPointer);
+      } else {
+        range = this.svcs.parserSvc.getRangeForJsonPath('asyncapi', jsonPointer);
+      }
+      
```

Comment:
```suggestion
      const methodName = doc.value.startsWith('asyncapi') ? 'getRangeForYamlPath' : 'getRangeForJsonPath';
      const range = this.svcs.parserSvc[methodName]('asyncapi', jsonPointer);
```

We can use chaining here for a concise code.

Reply:
Thanks @princerajpoot20 !
I've pushed the required changes regarding this.